crates/core/src/db/datastore/system_tables.rs

```diff
@@ -716,10 +751,419 @@ impl From<&StClientsRow> for ProductValue {
+            .next()
+        {
+            db.delete(tx, ST_VAR_ID, [row_ref.pointer()]);
+        }
+        db.insert(tx, ST_VAR_ID, ProductValue::from(StVarRow { name, value }))?;
+        Ok(())
+    }
+}
+
+/// A row in the system table `st_var`
+pub struct StVarRow {
+    pub name: StVarName,
+    pub value: StVarValue,
+}
+
+impl From<StVarRow> for ProductValue {
+    fn from(var: StVarRow) -> Self {
+        product!(var.name, var.value)
+    }
+}
+
+impl From<StVarRow> for AlgebraicValue {
+    fn from(row: StVarRow) -> Self {
+        AlgebraicValue::Product(row.into())
+    }
+}
+
+/// A system variable that defines a row limit for queries and subscriptions.
+/// If the cardinality of a query is estimated to exceed this limit,
+/// it will be rejected before being executed.
+pub const VARNAME_ROW_LIMIT: &str = "row_limit";
+/// A system variable that defines a threshold for logging slow queries.
+pub const VARNAME_SLOW_QUERY_MS: &str = "slow_ad_hoc_query_ms";
+/// A system variable that defines a threshold for logging slow subscriptions.
+pub const VARNAME_SLOW_SUB_MS: &str = "slow_subscription_query_ms";
+/// A system variable that defines a threshold for logging slow tx updates.
+pub const VARNAME_SLOW_INC_MS: &str = "slow_tx_update_ms";
+
+/// The name of a system variable in `st_var`
+#[derive(Debug, Clone, Copy)]
+pub enum StVarName {
+    RowLimit,
+    SlowQueryThreshold,
+    SlowSubThreshold,
+    SlowIncThreshold,
+}
+
+impl From<StVarName> for AlgebraicValue {
+    fn from(value: StVarName) -> Self {
+        match value {
+            StVarName::RowLimit => AlgebraicValue::String(VARNAME_ROW_LIMIT.to_string().into_boxed_str()),
+            StVarName::SlowQueryThreshold => AlgebraicValue::String(VARNAME_SLOW_QUERY_MS.to_string().into_boxed_str()),
+            StVarName::SlowSubThreshold => AlgebraicValue::String(VARNAME_SLOW_SUB_MS.to_string().into_boxed_str()),
+            StVarName::SlowIncThreshold => AlgebraicValue::String(VARNAME_SLOW_INC_MS.to_string().into_boxed_str()),
+        }
+    }
+}
+
+impl FromStr for StVarName {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            VARNAME_ROW_LIMIT => Ok(StVarName::RowLimit),
+            VARNAME_SLOW_QUERY_MS => Ok(StVarName::SlowQueryThreshold),
+            VARNAME_SLOW_SUB_MS => Ok(StVarName::SlowSubThreshold),
+            VARNAME_SLOW_INC_MS => Ok(StVarName::SlowIncThreshold),
+            _ => Err(anyhow::anyhow!("Invalid system variable {}", s)),
+        }
+    }
+}
+
+impl StVarName {
+    pub fn type_of(&self) -> StVarType {
+        match self {
+            StVarName::RowLimit
+            | StVarName::SlowQueryThreshold
+            | StVarName::SlowSubThreshold
+            | StVarName::SlowIncThreshold => StVarType::U64,
+        }
+    }
+}
+
+/// The type of a system variable in `st_var`
+#[repr(u8)]
+pub enum StVarType {
```

Comment:
Why do we need another `type` for this? I think just matching on `AlgebraicType` if wanna restrict to `primitive` types is enough

Reply:
Can a column be typed as `AlgebraicValue`? I was under the assumption it needed a concrete type. In this case, `AlgebraicType::Sum(...)`.

Reply:
We could conceivably `impl SpacetimeType for AlgebraicValue` by having an `AlgebraicType` which describes `AlgebraicValue`.